### PR TITLE
Fix complex tensor printing

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38031 Fix complex tensor printing**

Differential Revision: [D21502915](https://our.internmc.facebook.com/intern/diff/D21502915)

### CODE REVIEW

File: torch/_tensor_str.py [hunk 1]
```diff
@@ -280,7 +280,9 @@ def _str(self):
             or (self.device.type == 'cuda' and torch.cuda.current_device() != self.device.index):
         suffixes.append('device=\'' + str(self.device) + '\'')
 
-    has_default_dtype = self.dtype in (torch.get_default_dtype(), torch.int64, torch.bool)
+    # TODO: add an API to map real -> complex dtypes
+    _default_complex_dtype = torch.cfloat if torch.get_default_dtype() == torch.float else torch.cdouble
```
Comment (SsnL) [inline torch/_tensor_str.py:284]: so if default dtype is half this is cdouble?

Comment (anjali411) [reply torch/_tensor_str.py:284]: good point! no it should be cdouble if double else cfloat, similar to the logic [here](https://github.com/pytorch/pytorch/blob/a44824c9eda0301bb497840904ac1c64f0c69b89/c10/core/DefaultDtype.cpp#L11)

Comment (anjali411) [reply torch/_tensor_str.py:284]: will update!

**Review by ezyang — APPROVED** [review]
Test please? :)

**Pull request closed without merging**